src/Core/Framework/Api/OAuth/RefreshTokenRepository.php

```diff
@@ -32,31 +32,35 @@ public function getNewRefreshToken(): RefreshTokenEntityInterface
      */
     public function persistNewRefreshToken(RefreshTokenEntityInterface $refreshTokenEntity): void
     {
-        $this->connection->createQueryBuilder()
-            ->insert('refresh_token')
-            ->values([
-                'id' => ':id',
-                'user_id' => ':userId',
-                'token_id' => ':tokenId',
-                'issued_at' => ':issuedAt',
-                'expires_at' => ':expiresAt',
-            ])
-            ->setParameters([
-                'id' => Uuid::randomBytes(),
-                'userId' => Uuid::fromHexToBytes($refreshTokenEntity->getAccessToken()->getUserIdentifier()),
-                'tokenId' => $refreshTokenEntity->getIdentifier(),
-                'issuedAt' => (new \DateTime())->format(Defaults::STORAGE_DATE_TIME_FORMAT),
-                'expiresAt' => $refreshTokenEntity->getExpiryDateTime()->format(Defaults::STORAGE_DATE_TIME_FORMAT),
-            ])
-            ->executeStatement();
+        $userIdentifier = $refreshTokenEntity->getAccessToken()->getUserIdentifier();
```

Comment:
presumably this would never be falsy?  otherwise the token generation is bad, and it shouldn't have passed the verification with the key step. Perhaps an exception is more appropriate here 🤔

Reply:
It is intended, for authentication over integrations (not users) this is empty by design

Reply:
i added a comment, hopefully that makes it clearer